### PR TITLE
[Messenger][DX] Uses the adapter name instead of the service name

### PR DESCRIPTION
**Pull request by sroze**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ø
| License       | MIT
| Doc PR        | ø

As a developer, I don't get why I would need to use the service name while I give a name to the adapters. This is basically what this PR means for me in terms of configuration:
```patch
framework:
    messenger:
        routing:
            # We want the command below to be handled async with the "messenger.default_sender" (AMQP producer).
-            'App\Message\Command\CreateNumber': messenger.sender.amqp
+            'App\Message\Command\CreateNumber': amqp

        adapters:
            amqp: "%env(AMQP_DSN)%"
```

And in terms of pulling the messages:
```patch
- bin/console messenger:consume-messages messenger.receiver.amqp
+ bin/console messenger:consume-messages amqp
```

### CODE REVIEW

**Review by weaverryan — APPROVED** [review]
LGTM

File: src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php [hunk 1]
```diff
@@ -156,10 +157,22 @@ private function registerReceivers(ContainerBuilder $container)
...
 
         $container->getDefinition('messenger.receiver_locator')->replaceArgument(0, $receiverMapping);
     }
+
+    private function registerSenders(ContainerBuilder $container)
+    {
+        $senderLocatorMapping = array();
+        foreach ($container->findTaggedServiceIds('messenger.sender') as $id => $tags) {
+            foreach ($tags as $tag) {
+                $senderLocatorMapping[$tag['name'] ?? $id] = new Reference($id);
```
Comment (ogizanagi) [inline src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: Shouldn't it be available by both tag name and service id?

**Review by ogizanagi — APPROVED** [review]

File: src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php (hunk 1 above)
Comment (sroze) [reply src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: I don't think so actually. If it has a `name`, then that's what you use. It's aligned with what we discussed on other PRs regarding a single way of configuring/using the adapters.

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php [hunk 2]
```diff
@@ -544,8 +544,10 @@ public function testMessengerAdapter()
         $container = $this->createContainerFromFile('messenger_adapter');
         $this->assertTrue($container->hasDefinition('messenger.sender.default'));
         $this->assertTrue($container->getDefinition('messenger.sender.default')->hasTag('messenger.sender'));
+        $this->assertEquals(array(array('name' => 'default')), $container->getDefinition('messenger.sender.default')->getTag('messenger.sender'));
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php]: assertSame (same evrywhere)

File: src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: not allowing by-id forces knowing names, when knowing ids is easier sometimes (`debug:container --tag messenger.sender`)

**Review by nicolas-grekas — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php (hunk 2 above)
Comment (sroze) [reply src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php]: Updated for the valid ones (not for the ones containing `Reference`s...).

File: src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php (hunk 1 above)
Comment (sroze) [reply src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: Alright, you are two thinking that way and I'm mixed. Added, thanks 👍

**Comment by sroze** [comment]
Status: Needs review

**Review by fabpot — APPROVED** [review]

**Pull request merged**